site/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx

```diff
@@ -50,18 +52,45 @@ const CreateWorkspacePage: FC = () => {
       },
     },
   });
-  const {
-    template,
-    error,
-    parameters,
-    permissions,
-    gitAuth,
-    defaultName,
-    versionId,
-  } = createWorkspaceState.context;
+  const { template, parameters, permissions, defaultName, versionId } =
+    createWorkspaceState.context;
   const title = createWorkspaceState.matches("autoCreating")
     ? "Creating workspace..."
-    : "Create Workspace";
+    : "Create workspace";
+
+  const [gitAuthPollingState, setGitAuthPollingState] =
+    useState<GitAuthPollingState>("idle");
+
+  useEffect(() => {
+    if (gitAuthPollingState !== "polling") {
+      return;
+    }
+
+    // Poll for a maximum of one minute
+    const quitPolling = setTimeout(
+      () => setGitAuthPollingState("abandoned"),
+      60_000,
+    );
+    return () => {
+      clearTimeout(quitPolling);
+    };
+  }, [gitAuthPollingState]);
+
+  const startPollingGitAuth = useCallback(() => {
+    if (gitAuthPollingState === "polling") {
+      return;
+    }
+    setGitAuthPollingState("polling");
+  }, [gitAuthPollingState]);
```

Comment:
that's not the intended behavior. the idea is to wait until we know there's a chance that something is worth polling.

Reply:
I feel like I said this weird. What I was trying to say is that if the goal is to make sure that React does nothing at all in the off chance that the callback is called while polling is already happening, `setGitAuthPollingState("polling")` will do that.

The function will still dispatch a new value, but because it would be the same value as before (especially with the state being a primitive), React rejects the state change and doesn't re-render

Reply:
oh, sorry 😅 I think you're right